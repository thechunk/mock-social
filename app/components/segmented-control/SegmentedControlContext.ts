import {Context, createContext} from "react";

const SegmentedControlContext: Context<ISegmentedControlContext> = createContext({
    selectedTab: 0,
    setSelectedTab: (id: number) => {}
});
export default SegmentedControlContext;