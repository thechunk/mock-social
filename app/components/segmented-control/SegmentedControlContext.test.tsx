import * as React from 'react';
import SegmentedControlContext from "./SegmentedControlContext";

describe('SegmentedControlContext', () => {
    it('creates context', () => {
        expect(SegmentedControlContext.Provider).not.toBeUndefined();
        expect(SegmentedControlContext.Consumer).not.toBeUndefined();
    });

    it('contains context values', () => {
        expect(<SegmentedControlContext.Provider value={{selectedTab: 1, setSelectedTab: () => {}}} />)
            .toMatchSnapshot();
    })
});
