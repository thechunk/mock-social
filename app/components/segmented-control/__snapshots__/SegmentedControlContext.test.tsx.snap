// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControlContext contains context values 1`] = `
<Context.Provider
  value={
    Object {
      "selectedTab": 1,
      "setSelectedTab": [Function],
    }
  }
/>
`;

exports[`SegmentedControlContext contains default context values 1`] = `<Context.Provider />`;
